src/client.rs

```diff
@@ -194,7 +190,16 @@ impl LastFm {
         params.insert("method".to_string(), operation.to_string());
         params.insert("api_sig".to_string(), signature);
 
-        self.http_client.post(url).form(&params).send()
+        let params: Vec<(&str, &str)> = params
+            .iter()
+            .map(|(k, v)| (k.as_str(), v.as_str()))
+            .collect();
+
+        let resp = self.http_client.post(url).send_form(&params[..]);
+        match resp.synthetic_error() {
```

Comment:
`ureq` always returns a response object, even when there is a transfer error ([docs](https://docs.rs/ureq/1.5.4/ureq/#synthetic-errors)). This small check unmasks transfer errors masquerading as valid responses.